### DISCUSSION TITLE
Issues Accessing Test Set Dataloader in tsai

### DISCUSSION THREAD

**Discussion by Loccret** [timeseriesAI/tsai#770] [Q&A] [answered]
Hello everyone,

I am encountering some confusion regarding the test set dataloader in tsai. I have created my `dls` (data loaders) using the following code:

```python
import tsai.all as ta

C, label, splits = ta.combine_split_data([C_train, C_valid, C_test], 
                                        [label_train, label_valid, label_test])
tfms  = [None, [ta.TSClassification()]]
batch_tfms = [ta.TSStandardize(by_sample=True, by_var=False)]

device = torch.device('cuda:0')
dls = ta.get_ts_dls(C, label, splits=splits, tfms=tfms, test_items=C_test,
                batch_tfms=batch_tfms, bs=10000, shuffle_train=True, device=device)
```
I need to extract rocket features, and therefore I require access to `dls.train`, `dls.valid`, and `dls.test` separately. However, when attempting to access `dls.test`, I receive an error message:
```Python
AttributeError: test
```
However, when training other models such as `Transformer` and `InceptionTime`, after fitting, `learn.get_preds(ds_idx = 2)` can access the test data correctly. Therefore, I know that the test dataloader is indeed included in the `dls`. However, I do not know how to access this test dataloader.

Can someone please help me with this issue?

Thank you in advance!

**Comment by vrodriguezf** [comment]
Does `learn.dls[2]`work?

**Reply by Loccret** [reply]
Hey @vrodriguezf .

Thank you for your reply.

I'm currently working with `ROCKET` to determine if there are any performance improvements, and I'm following the example provided in the documentation which can be found [here](https://timeseriesai.github.io/tsai/models.rocket_pytorch.html).

It appears that no `learner` has been created.


I would like get something like:
```Python
X_test, y_test = create_rocket_features(dls.test, model)
```

Unfortunately, I'm receiving an error message which reads:
```Python
AttributeError: test
```

Would you please tell me how to do it?

**Comment by Loccret** [comment, marked as answer]
This is a bug that was reported in #724.

In the latest version (0.3.4), the test set can be accessed correctly by using `dls[2]`.
